### PR TITLE
LDrawLoader example: Add option to merge geometries after loading

### PR DESCRIPTION
**Pull request by yomboprime**
Related comment: https://github.com/mrdoob/three.js/pull/23168#issuecomment-1007111263

**Description**

In the LDrawLoader example, add a gui option (and function) to merge model geometries by material after model loading. The option is named 'Merge model' and gives better render performance. This option disables the 'Construction steps' and the 'Show conditional lines' options.

The results are more noticeable in Firefox, where frames are almost not dropped where they dropped before even on small models. The model process time is somewhat 1/5 of loading time.

The function ```mergeObject()``` is 180 lines out of 560 of the example and it is fairly complex, so I'd rather move it to the LDrawLoader. But this can be done at any time.

### CODE REVIEW

**Comment by yomboprime** [comment]
Umm I think I had my git repo a bit dirty. Only the last commit of those 11 is relevant for this PR.

@gkjohnson Is it Ok or should I clean my repo and make another PR?

**Comment by yomboprime** [comment]
I've enabled again the conditional lines hide option because I've seen it could be done easily.

File: examples/webgl_loader_ldraw.html
Comment (gkjohnson) [inline examples/webgl_loader_ldraw.html:367]: I don't think theres any reason to permute the conditional lines, is there? The only reason the faces need to be adjusted is because of winding order which doesn't apply here.

**Review by gkjohnson — COMMENTED** [review]
Awesome! This is a nice utility function and keeps the LDrawLoader more clean.

In the previous merge objects code path the merged geometry resulted in one mesh with different groups and a material array (in addition to the two line segment objects) which simplifies the final object. It might be nice but I don't necessarily have strong opinions on it!

Live demo:

https://raw.githack.com/yomboprime/three.js/ldraw_merge_model/examples/webgl_loader_ldraw.html

**Comment by gkjohnson** [comment]
> @gkjohnson Is it Ok or should I clean my repo and make another PR?

This is more a question for @mrdoob. The PR is easy enough to look at on its own for me and the commits will be squashed on merge.

File: examples/webgl_loader_ldraw.html
Comment (yomboprime) [reply examples/webgl_loader_ldraw.html:367]: I was not sure about that, I'll change it.

**Comment by yomboprime** [comment]
@mrdoob The PR is ready to merge if those old commits are ok to be squashed.

**Review by gkjohnson — APPROVED** [review]

File: examples/webgl_loader_ldraw.html
Comment (mrdoob) [inline examples/webgl_loader_ldraw.html:253]: What do you guys think about moving this code into a static method on `LDrawLoader` and keep the example code simple?

Comment (yomboprime) [reply examples/webgl_loader_ldraw.html:253]: Yes, I was thinking the same. The loader code has been simplified by extracting this functionality to the function. I will do it in some hours.

Comment (yomboprime) [reply examples/webgl_loader_ldraw.html]: Done!

**Comment by yomboprime** [comment]
Done, ready to merge.

File: examples/jsm/loaders/LDrawLoader.js
Comment (gkjohnson) [inline examples/jsm/loaders/LDrawLoader.js:21]: Small note that this is maybe a bit related to #23167 now that it's importing a utils function and is a breaking change for anyone using the traditional global import approach.

**Review by gkjohnson — APPROVED** [review]

File: examples/jsm/loaders/LDrawLoader.js
Comment (yomboprime) [reply examples/jsm/loaders/LDrawLoader.js:21]: Yes, definitely it is related. Is there anymore that can be done in this PR? Could the utility function ```mergeObject``` be placed in another class to not disturb LDrawLoader?

Comment (gkjohnson) [reply examples/jsm/loaders/LDrawLoader.js:21]: I think it's okay as-is. Was just pointing it out. Maybe it should be added to the migration guide?

Comment (yomboprime) [reply examples/jsm/loaders/LDrawLoader.js:21]: I'd feel more comfortable by creating a new file ```jsm/utils/LDrawUtils.js``` and moving the function there. I'd also create a documentation file for LDrawLoader with explanations on the loader, the ```LDrawUtils``` and the ```packLDrawModel``` utility. Let me know if you agree @gkjohnson @mrdoob

Comment (gkjohnson) [reply examples/jsm/loaders/LDrawLoader.js:21]: I'll defer to @mrdoob!

Comment (mrdoob) [reply examples/jsm/loaders/LDrawLoader.js:21]: `LDrawUtils` sounds good 👍

**Comment by yomboprime** [comment]
I made the ```LDrawUtils.js``` and the doc file for ```LDrawLoader```.

You can see the new doc page here:
https://raw.githack.com/yomboprime/three.js/ldraw_merge_model/docs/index.html#examples/en/loaders/LDrawLoader

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**
